src/VASSAL/build/module/Chatter.java

```diff
@@ -115,55 +132,214 @@ public void actionPerformed(ActionEvent e) {
     return input;
   }
 
+  //BR// To make the player colors easy to override in a custom class 
+  //BR// (my modules have logic to assign individual player colors -- beyond the scope of the present effort but a perhaps a fun future addition)
+  protected String getChatStyle(String s)
+  {   
+    if (s.startsWith(formatChat("").trim())) { //$NON-NLS-1$
+      return "mychat";      
+    } else {
+      return "other";
+    }   
+  }
+
   /**
    * Display a message in the text area
    */
   public void show(String s) {
-    conversation.append("\n" + s); //$NON-NLS-1$
+    String style;
+    boolean html_allowed;
+
+    //BR// Choose an appropriate style to display this message in
+    s = s.trim();
+    if (!s.isEmpty()) {
+      if (s.startsWith("*")) {
+        //BR// Here we just extend the convention of looking at first characters, this time to the second character.
+        //BR// | = msg1 (w/ HTML parsing explicitly opted in)
+        //BR// ! = msg2
+        //BR// ? = msg3
+        //BR// ~ = msg4
+        //BR// ` = msg5
+        //BR// These characters can be pre-pended to Report messages to produce the color changes. The characters themselves are removed before display.
+        //BR// Reports can also include <b></b> tags for bold and <i></i> for italic.
+        if (s.startsWith("* |") || s.startsWith("*|")) {
+          style = "msg";
+          s = s.replaceFirst("\\|", "");
+          html_allowed = true;
+        } else if (s.startsWith("* !") || s.startsWith("*!")) {
+          style = "msg2";
+          s = s.replaceFirst("!", "");
+          html_allowed = true;
+        } else if (s.startsWith("* ?") || s.startsWith("*?")) {
+          style = "msg3";
+          s = s.replaceFirst("\\?", "");
+          html_allowed = true;
+        } else if (s.startsWith("* ~") || s.startsWith("*~")) {
+          style = "msg4";
+          s = s.replaceFirst("~", "");
+          html_allowed = true;
+        } else if (s.startsWith("* `") || s.startsWith("*`")) {
+          style = "msg5";
+          s = s.replaceFirst("`", "");
+          html_allowed = true;
+        } else {
+          style = "msg";
+          html_allowed = !htmlCompatibility; //BR// Generic report lines check compatibility flag (so old modules will not break on e.g. "<" in messages)
+        }
+      } else if (s.startsWith("-")) {
+        style = "sys";
+        html_allowed = true;
+      } else {
+        style = getChatStyle(s);
+        html_allowed = false;
+      }
+    } else {
+      style = "msg";
+      html_allowed = false;
+    }
+
+    //BR// Disable unwanted HTML tags in contexts where it shouldn't be allowed:
+    //BR// (1) Anything received from chat channel, for security reasons
+    //BR// (2) Legacy module "report" text when not explicitly opted in w/ first character or preference setting 
+    if (!html_allowed) {
+      s = s.replaceAll("<", "&lt;"); //BR// This prevents any unwanted tag from functioning
```

Comment:
Can chain these and format like this, with the comments:
```
s = s.replaceAll(a, b) // comment 1
     .replaceAll(c, d) // comment 2
```

Reply:
fixed for next push